### PR TITLE
feat!: Refactor workflows and add flags to generate steps

### PR DESCRIPTION
**Pull request by mariusz89016**
Refactor workflows and add flags to generate steps for running Checkov, installing packages with asdf, pulling GitLab variables & checking allowed approvers.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -65,24 +65,79 @@ variable "workflows" {
   description = "List of custom workflow that will be added to the repo config file"
   type = map(object({
     plan = optional(object({
```
Comment (mariusz89016) [inline variables.tf:67]: todo: validation

File: workflow_templates.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
+locals {
+  null_workflow = {
```
Comment (mariusz89016) [inline workflow_templates.tf:2]: let me know if there's better/easier way to do this

File: main.tf [hunk 3]
```diff
@@ -34,23 +34,64 @@ locals {
...
+          for step in stage.steps : [
+            for name, object in step :
+            jsondecode(
+              name == "atlantis_step" ?
+              (object.extra_args != null ? jsonencode({ (object.command) : { extra_args : object.extra_args } }) : jsonencode(object.command)) :
+              jsonencode({ (name) : object })
+            )
+            if object != null
+        ]]),
+        [for e in ["show", { run = format("checkov -f $SHOWFILE -o github_failed_only %s", workflow.checkov.soft_fail ? "--soft-fail" : "") }] : e
```
Comment (mariusz89016) [inline main.tf:80]: for now I decided to do soft-fail flag inline but in the future it'd be better to maybe make submodule out of that...?

**Review by jakubigla — APPROVED** [review]
LGTM, but please fix type in the first paragraph in the readme file

**Pull request merged**